sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java

```diff
@@ -220,8 +220,11 @@ private ShardLocation getShardLocation(String indexName, String id, @Nullable St
                 nodeId = shardRouting.currentNodeId();
             }
 
-            if (nodeId == null && LOGGER.isDebugEnabled()) {
-                LOGGER.debug("Unable to get the node id for index {} and shard {}", indexName, id);
+            if (nodeId == null) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Unable to get the node id for index {} and shard {}", indexName, shardIterator.shardId());
+                }
+                return null;
```

Comment:
why are you returning null? Looking at createRowConsumer if the shardLocation is null we will add to pending requests and not to requests. Is that by design? If it is we need to document this somehow (it's a side effect now and not really sure it's correct).